### PR TITLE
Axis par args not getting into the render

### PR DESCRIPTION
**Pull request by jordansread**
tcl and others were being ignored in `axis`. This is aimed at fixing #456

### CODE REVIEW

File: R/axis.R [hunk 1]
```diff
@@ -68,28 +68,39 @@ axis.gsplot <- function(object, ..., n.minor=0, tcl.minor=0.15, reverse=NULL, ap
...
-                                custom.config = object[["global"]][["config"]][["config.file"]])$call.args
+  args <- filter_arguments(fun.name = fun.name, ..., 
+                           custom.config = object[["global"]][["config"]][["config.file"]])
   
-  sides <- user_args[[fun.name]]$side
-  user_args[[fun.name]]$side <- NULL
-  user_args[[fun.name]]$n.minor <- n.minor
-  user_args[[fun.name]]$tcl.minor <- tcl.minor
+  if (length(args$extracted.args) > 0){
+    warning('axs.gsplot is not configured to support embedded function calls', call. = FALSE)
```
Comment (lindsayplatt) [inline R/axis.R]: warning message should say "axis" not "axs"

File: R/axis.R [hunk 2]
```diff
@@ -68,28 +68,39 @@ axis.gsplot <- function(object, ..., n.minor=0, tcl.minor=0.15, reverse=NULL, ap
...
+  
+  if (is.null(sides)){
+    return(object)
+  }
+  
+  user.args[[fun.name]]$side <- NULL
+  user.args[[fun.name]]$n.minor <- n.minor
+  user.args[[fun.name]]$tcl.minor <- tcl.minor
+  
+  user.args[[fun.name]] <- append_replace(user.args[[fun.name]], args$option.args)
```
Comment (lindsayplatt) [inline R/axis.R:89]: correct me if I'm wrong, but this is the legitimate change for this PR, right? The others are adding warnings, catching nulls, and changing code style/object names, but this line is what has been actually added

File: tests/testthat/tests-axis.R [hunk 3]
```diff
@@ -83,6 +83,21 @@ test_that("axis tracks append FALSE by default",{
...
+})
+
+test_that("special args given to axis are retained", {
+  gs <- points(gsplot(), 1, 0) %>% axis(side=1, n.minor = 4)
+  expect_equal(gs$side.1$axis[["n.minor"]], 4)
+  
+  gs <- points(gsplot(), 1, 0) %>% axis(side=1, tcl.minor = -0.136)
+  expect_equal(gs$side.1$axis[["tcl.minor"]], -0.136)
+})
+
```
Comment (lindsayplatt) [inline tests/testthat/tests-axis.R:100]: should we add a test that does something with axis style args in other places? `gs <- points(gsplot(), 1, 0, tcl=-0.4) %>% axis(side=1, n.minor = 4)`

Comment (jordansread) [reply tests/testthat/tests-axis.R:100]: I can add that.

File: R/axis.R (hunk 2 above)
Comment (jordansread) [reply R/axis.R:89]: yes, this is the meat of it

File: tests/testthat/tests-axis.R (hunk 3 above)
Comment (jordansread) [reply tests/testthat/tests-axis.R:100]: @lindsaycarr added things per your review

Comment (lindsayplatt) [reply tests/testthat/tests-axis.R:100]: yep, waiting for travis to finish and then I will merge!

**Pull request merged**